### PR TITLE
Tbittner - Game of Life Solution

### PR DESCRIPTION
**Pull request by FGS-TBittner**
`GameOfLife` will load a `Game` from a file based on the input/output specified in the GameOfLife Kata. The `Game` will then be displayed on the `Console` and will proceed through generations as `Enter` is pressed until `Q` or `q` is entered to quit. `GameOfLife` takes two arguments: the path of the file to load, and if the `Game` should wrap `Grid` edges. If no second parameter is provided, the value is assumed to be `false`. As the `Game` progresses, `Cells` will live or die based on the rules specified by Conway's Game of Life.

### CODE REVIEW

File: GameOfLife.Tests/CellEnvironmentAnalyzerTests.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+using NUnit.Framework;
+using Ploeh.AutoFixture;
+using System.Linq;
+
+namespace GameOfLife.Tests
+{
+    [TestFixture]
+    public class CellEnvironmentAnalyzerTests : TestsUsingGrid
+    {
+        private const int MaximumNeighboursToLive = CellEnvironmentAnalyzer.MaximumNeighboursToLive;
```
Comment (FGS-TBittner) [inline GameOfLife.Tests/CellEnvironmentAnalyzerTests.cs]: Just trying to class it up!
It was the spelling they used in the Kata description and I didn't really think about it.

File: GameOfLife.Console/GameOfLife.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
+﻿using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace GameOfLife.Console
+{
+    public class GameOfLife
```
Comment (fgsahoward) [inline GameOfLife.Console/GameOfLife.cs]: Much better. Consider adhering to the [Inversion of Control Principle](https://martinfowler.com/articles/injection.html) for all these `static` classes you have extracted. That should make your program more modular, and more resilient to change. If you're feeling up to it after you have refactored with this in mind, you may even wish to incorporate an IoC container, but I won't make that a requirement of this.

File: GameOfLife.Console/ArgumentParser/CommandLineArgumentParser.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    public class CommandLineArgumentParser : IArgumentParser
+    {
+        private readonly string[] _arguments;
+
+        public CommandLineArgumentParser(string[] arguments)
+        {
+            _arguments = arguments;
+        }
+
+        public string GetFilePathArgument() => _arguments.Length > 0 ? _arguments[0] : "";
```
Comment (fgsahoward) [inline GameOfLife.Console/ArgumentParser/CommandLineArgumentParser.cs]: Just a note for the future, stylecop will complain about `""` unless you use `string.Empty`

**Comment by fgsahoward** [comment]
image

**Pull request merged**